apps/btc_family/dash/dash_app.c

```diff
@@ -0,0 +1,133 @@
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "dash_app.h"
+
+#include "btc_helpers.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Check if the purpose index is supported by the Dash app.
+ * @details The current supported/allowed purpose index is 0x8000002C
+ *
+ * @param purpose_index The purpose index to be checked
+ *
+ * @return bool Indicates if the provided purpose index is supported
+ * @retval true The provided purpose index is supported
+ * @retval false The provided purpose index is not supported
+ */
+static bool is_purpose_supported(uint32_t purpose_index);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+const btc_config_t dash_app = {
+    .coin_type = COIN_DASH,
+    .p2pkh_addr_ver = 0x4c,
+    .p2sh_addr_ver = 0x10,
+    .legacy_xpub_ver = 0x0488b21e,
+    .segwit_xpub_ver = 0x00,
+    .nsegwit_xpub_ver = 0x0488b21e,
```

Comment:
Native segwit format is not supported for dash, please re-check this configuration

Reply:
corrected